src/autopas/particles/ParticleBase.h

```diff
@@ -169,7 +169,7 @@ class ParticleBase {
     const auto distanceVec = r - _r;
     const double distanceSquared = utils::ArrayMath::dot(distanceVec, distanceVec);
     setR(r);
-    const bool distanceIsFine = distanceSquared < maxDistSquared;
+    const bool distanceIsFine = distanceSquared < maxDistSquared or maxDistSquared == 0;
```

Comment:
I'm concerned about the use of `==`. As `maxDistSquared` is a calculated value in md-flexible (as opposed to just being assigned to zero), with might not exactly be zero even if it is supposed to be. Even if we could forcibly assign this as zero in the case of md-flexible, we couldn't do this for every code. I would use `isNearAbs` or, if you are worried about the cost, a simple `maxDistSquared < 1e-12` is probably enough(?).

Reply:
@SamNewcome @FG-TUM 
I want to go with 'maxDistSquared < 1e-12' because I am comparing to zero, I am confident it is positive (square) and it would be slightly faster if number of particles in simulation is large. Is this fine?

I could use `isNearAbs()` alternatively.

Reply:
I'm fine with using a simple `maxDistSquared < 1e-12`. I'm slightly concerned that 1e-12 is not necessarily always going to be suitable. E.g. if the simulation was using units/scales such that 1e-12 was actually an expected amount to move in a timestep. I am however somewhat okay with this in the context that we aren't going to do anything like this in the short term and in the long term we hope to replace this mechanic.

Reply:
Ok, fair point :) done!